### PR TITLE
feat(halo): replace OmniStake with Staking in evmengine

### PR DESCRIPTION
**Pull request by kevinhalliday**
Replace OmniStake.sol with Staking.sol in evmengine

task: https://app.asana.com/0/1206208509925075/1207375377017417

### CODE REVIEW

File: halo/evmstaking/evmstaking.go [hunk 1]
```diff
@@ -26,13 +27,19 @@ import (
 	stypes "github.com/cosmos/cosmos-sdk/x/staking/types"
 )
 
-const AccountName = "evmstaking"
+const ModuleName = "evmstaking"
```
Comment (kevinhalliday) [inline halo/evmstaking/evmstaking.go:30]: ModuleName is standard no?

File: halo/evmstaking/evmstaking.go [hunk 2]
```diff
@@ -91,73 +98,81 @@ func (p EventProcessor) Prepare(ctx context.Context, blockHash common.Hash) ([]*
...
 
-	pubkey, err := k1util.StdPubKeyToCosmos(stdPubkey)
+// deliverCreateValidator processes a CreateValidator event, and creates a new validator.
+// - Mint the corresponding amount of $STAKE coins.
+// - Send the minted coins to the depositor's account.
+// - Create a new validator with the depositor's account.
+//
+// NOTE: if we error before coins are sent, the deposit is lost.
+// NOTE: if we error after coins are sent, the account will have $STAKE tokens minted to it, with no way of accessing.
+// TODO: consider recovery methods for the above scenarios.
```
Comment (kevinhalliday) [inline halo/evmstaking/evmstaking.go:141]: These cases worth discussing

Comment (corverroos) [reply halo/evmstaking/evmstaking.go:141]: errors result in rollback of all updated state, so nothing to worry about, any error results in no state updates

**Review by corverroos — APPROVED** [review]

File: halo/evmstaking/evmstaking.go (hunk 2 above)
Comment (kevinhalliday) [reply halo/evmstaking/evmstaking.go:141]: oh sweet

**Pull request merged**